modules/imgproc/test/test_resize_bitexact.cpp

```diff
@@ -18,140 +18,152 @@ namespace opencv_test { namespace {
+            for (int c = 0; c < cn; c++)
             {
-                softdouble src_row_flt = scale_y*(softdouble(j) + softdouble(0.5)) - softdouble(0.5);
-                int src_row = cvFloor(src_row_flt);
-                int64_t ycoeff1 = cvRound64((src_row_flt - softdouble(src_row))*softdouble(fixedOne));
-                int64_t ycoeff0 = fixedOne - ycoeff1;
-
-                for (int i = 0; i < dcols; i++)
-                {
-                    softdouble src_col_flt = scale_x*(softdouble(i) + softdouble(0.5)) - softdouble(0.5);
-                    int src_col = cvFloor(src_col_flt);
-                    int64_t xcoeff1 = cvRound64((src_col_flt - softdouble(src_col))*softdouble(fixedOne));
-                    int64_t xcoeff0 = fixedOne - xcoeff1;
-
-                    uint8_t* dst_pt = refdst.ptr(j, i);
-                    uint8_t* src_pt00 = src.ptr( src_row      < 0 ? 0 :  src_row      >= rows ? rows - 1 :  src_row     ,
-                                                 src_col      < 0 ? 0 :  src_col      >= cols ? cols - 1 :  src_col     );
-                    uint8_t* src_pt01 = src.ptr( src_row      < 0 ? 0 :  src_row      >= rows ? rows - 1 :  src_row     ,
-                                                (src_col + 1) < 0 ? 0 : (src_col + 1) >= cols ? cols - 1 : (src_col + 1));
-                    uint8_t* src_pt10 = src.ptr((src_row + 1) < 0 ? 0 : (src_row + 1) >= rows ? rows - 1 : (src_row + 1),
-                                                 src_col      < 0 ? 0 :  src_col      >= cols ? cols - 1 :  src_col     );
-                    uint8_t* src_pt11 = src.ptr((src_row + 1) < 0 ? 0 : (src_row + 1) >= rows ? rows - 1 : (src_row + 1),
-                                                (src_col + 1) < 0 ? 0 : (src_col + 1) >= cols ? cols - 1 : (src_col + 1));
-                    for (int c = 0; c < cn; c++)
-                    {
-                        if (depth == CV_8U)
-                            eval4< uint8_t, fixedShiftU8>(xcoeff0, xcoeff1, ycoeff0, ycoeff1, c, src_pt00, src_pt01, src_pt10, src_pt11, dst_pt);
-                        else if (depth == CV_16U)
-                            eval4<uint16_t, fixedShiftU8>(xcoeff0, xcoeff1, ycoeff0, ycoeff1, c, src_pt00, src_pt01, src_pt10, src_pt11, dst_pt);
-                        else if (depth == CV_16S)
-                            eval4< int16_t, fixedShiftU8>(xcoeff0, xcoeff1, ycoeff0, ycoeff1, c, src_pt00, src_pt01, src_pt10, src_pt11, dst_pt);
-                        else if (depth == CV_32S)
-                            eval4< int32_t, fixedShiftU8>(xcoeff0, xcoeff1, ycoeff0, ycoeff1, c, src_pt00, src_pt01, src_pt10, src_pt11, dst_pt);
-                        else
-                            CV_Assert(0);
-                    }
-                }
+                if ((depth == CV_8U) || (depth == CV_Bool))
+                    eval4< uint8_t, fixedShiftU8>(xcoeff0, xcoeff1, ycoeff0, ycoeff1, c, src_pt00, src_pt01, src_pt10, src_pt11, dst_pt);
```

Comment:
I believe we should check that all values in the result image are either `0` or `255` for `CV_Bool`.